src/lib/components/player/presence.svelte

```diff
@@ -0,0 +1,165 @@
+<script lang="ts">
+   import { fade, slide } from 'svelte/transition';
+
+   import { State, userPresence } from '$lib/stores/presence-store';
+   import { searchView } from '$lib/stores/global-store';
+
+   import { CDN_URL } from '$lib/utils/env';
+   import { getDifficultyLabelSmall, getDifficultyStyle } from '$lib/utils/helpers';
+   import { GameMode } from '$lib/utils/presence-socket';
+
+   export let playerId: string | bigint;
+
+   let status: ReturnType<typeof userPresence>;
+   $: status = userPresence(playerId);
+
+   const openSearch = (val: string) => {
+      $searchView?.setVisibility(true);
+      $searchView?.search(val);
+   };
+</script>
+
+<div class="status-container">
+   {#key playerId}
+      {#if $status}
+         {@const isPlaying = $status.state === State.Playing && $status.currentMap.mode !== GameMode.Practice}
+         <div class="window-header statusbar" class:transparent={!isPlaying} transition:slide>
+            <div class="dot" class:green={$status.state !== State.Offline} />
+            {#key $status.state === State.Playing ? $status.currentMap.mode : $status.state}
+               <div class="status-text" transition:slide>
+                  {#if $status.state === State.Offline}
+                     Offline
+                  {/if}
+                  {#if $status.state === State.Menu}
+                     Browsing Menus
+                  {/if}
+                  {#if $status.state === State.Online}
+                     Online
+                  {/if}
+
+                  {#if $status.state === State.Playing}
+                     {#if $status.currentMap.mode !== GameMode.Practice}
+                        Playing Beat Saber
+                     {:else}
+                        In Practice Mode
+                     {/if}
+                  {/if}
+               </div>
+            {/key}
+         </div>
+      {:else}
+         <div class="window-header statusbar transparent" transition:slide>
+            <img src="/images/loading.svg" alt="Loading..." class="loading" />Connecting...
+         </div>
+      {/if}
+      {#if $status && $status.state === State.Playing && $status.currentMap && $status.currentMap.mode !== GameMode.Practice}
+         {#key $status.currentMap}
+            <div class="window" transition:slide>
+               <div class="ingame-info">
+                  <div class="song-cover">
```

Comment:
We should probably re-use the already established component for this.

But this part of the PR is blocked until modifying the leaderboard page to accept a hash difficulty id combo, a redirect would likely be ideal.

Reply:
I tried doing that but the existing component is pretty tightly coupled to the score row, and this is pretty simple

i'm also planning to add some more information beyond map info around christmas when i have more time, so that is also a factor

Reply:
That's sad, if you're up to it making that component reusable would be nice, but we are throwing away this codebase soon so meh.